lib/web_ui/lib/src/engine/assets.dart

```diff
@@ -58,26 +57,34 @@ class AssetManager {
     return Uri.encodeFull((_baseUrl ?? '') + '$assetsDir/$asset');
   }
 
-  /// Loads an asset using an [html.HttpRequest] and returns data as [ByteData].
+  /// Loads an asset using an [DomXMLHttpRequest] and returns data as [ByteData].
   Future<ByteData> load(String asset) async {
     final String url = getAssetUrl(asset);
     try {
-      final html.HttpRequest request =
-          await html.HttpRequest.request(url, responseType: 'arraybuffer');
+      final DomXMLHttpRequest request =
+          await domHttpRequest(url, responseType: 'arraybuffer');
 
       final ByteBuffer response = request.response as ByteBuffer;
       return response.asByteData();
-    } on html.ProgressEvent catch (e) {
-      final html.EventTarget? target = e.target;
-      if (target is html.HttpRequest) {
-        if (target.status == 404 && asset == 'AssetManifest.json') {
+    } catch (e) {
+      if (!domInstanceOfString(e, 'ProgressEvent')){
```

Comment:
Hmmm... a lot to unpack here. Seems to me like all the special casing around `ProgressEvent` is not necessary anymore, since the code you've written in `DomHttpRequest` doesn't actually listen for progress events at all and won't raise any, from my reading of the code. It seems like the weirdness from the previous code was maybe part of a clumsy attempt to jam everything (including progress messages) into one async promise/future call. So I guess some questions that come to mind:
1) Is tracking progress necessary at all? If not, we can probably just ditch this code path and the try/catch block completely.
2) If it is necessary, rather than propagating the progress events via the promise/future itself (which is pretty 🤨 ) could we maybe just allow the client to optionally pass in a progress handler that gets invoked with progress events separately from the promise/future?

Reply:
After further reading, I'm realizing now that an error is propagated with a type of `ProgressEvent` in JS. So I guess disregard this comment, I see what's going on now.